python/opae.admin/opae/admin/fpga.py

```diff
@@ -442,34 +441,55 @@ def supports_rsu(self):
         """
         return self.pci_node.pci_id in self.BOOT_PAGES
 
-    def rsu_boot(self, page, **kwargs):
+    def rsu_boot(self, factory, **kwargs):
         # look for non-max10 solution
         fme = self.fme
         if fme:
             bmc = fme.avmmi_bmc
             if bmc is None:
-                self._rsu_boot_sysfs(page, **kwargs)
+                self._rsu_boot_sysfs(factory, **kwargs)
             else:
                 bmc.reboot()
         else:
             self.log.warn('do not have FME device')
 
-    def _rsu_boot_sysfs(self, page, **kwargs):
+    def _rsu_boot_sysfs(self, factory, **kwargs):
         boot_type = kwargs.pop('type', 'bmcimg')
+
         if kwargs:
             self.log.exception('unrecognized kwargs: %s', kwargs)
             raise ValueError('unrecognized kwargs: {}'.format(kwargs))
         if boot_type not in self.BOOT_TYPES:
             raise TypeError('type: {} not recognized'.format(boot_type))
 
-        if boot_type == "bmcimg":
-            boot_type = "bmc"
-        node_path = ("m10bmc-secure.*.auto/ifpga_sec_mgr/"
-                     "ifpga_sec*/update/{boot_type}_image_load").format(
-                         boot_type=boot_type)
+        ifpga_sec = self.secure_dev
+        if not ifpga_sec:
+            msg = 'rsu not supported by this (0x{:04x},0x{:04x})'.format(
+                self.pci_node.pci_id[0], self.pci_node.pci_id[1])
+            self.log.exception(msg)
+            raise TypeError(msg)
+
+        available_images = ifpga_sec.find_one('update/available_images').value
+        image_load = ifpga_sec.find_one('update/image_load')
 
-        node = self.fme.spi_bus.find_one(node_path)
-        node.value = page
+        if boot_type == 'bmcimg':
+            if factory:
```

Comment:
I'm not a python expert...
If factory an integer value, containing the value passed to the tool (0 or 1). Unfortunately, the values for factory/user are swapped between VC and DC. It is not clear to me if this logic is taking that into acount.

Reply:
Good call.  Actually, factory is being passed in from the top-level command option (-f) to the rsu script.
https://github.com/OPAE/opae-sdk/pull/1653/files#diff-8a2882609782f926ff14ff3440bf3167R153